source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,328 @@
+
+       [BsonConstructor]
+       public Person(string name, string age)
+       {
+           Name = name;
+           Age = age;
+       }
+   }
+
+.. tip:: Multiple ``BsonConstructor`` attributes
+
+   If there is more than one constructor with the ``BsonConstructor``
+   attribute, the driver uses the constructor that has the most
+   parameters with matching fields in the document.
+
+You can also specify the constructor to use when registering your class map:
+
+.. code-block:: csharp
+   :emphasize-lines: 17
+
+   public class Person 
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+
+       public Person(string name, string age)
+       {
+           Name = name;
+           Age = age;
+       }
+   }
+
+   BsonClassMap.RegisterClassMap<Person>(classmap =>
+   {
+       classmap.AutoMap();
+       classmap.MapCreator(p => new Person(p.Name, p.Age));
+   });
+
+Customize Property Serialization
+--------------------------------
+
+You can customize how the driver serializes a class property by
+adding attributes to the property. For more information about custom
+property serialization, see :ref:`csharp-custom-serialization`.
+
+Support Extra Elements
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can design your {+language+} class to store any extra elements in your
+document that don't have matching class properties. To do this your class must
+have a ``BsonDocument`` type property to hold the extra elements.
+
+The following code uses the ``BsonExtraElements`` attribute with the
+``ExtraElements`` property to direct the driver to store extra elements:
+
+.. code-block:: csharp
+
+   public class Person
+   {
+       public string Name { get; set; 
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+       [BsonExtraElements]
+       public BsonDocument ExtraElements {get; set;}
+   }
+
+You can also support extra elements when initializing a class map as follows:
+
+.. code-block:: csharp
+   
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.MapExtraElementsMember(p => p.ExtraElements);
+   });
+
+.. note::
+
+   After the driver serializes a class with extra elements back to BSON, the 
+   extra elements may not be in the same order as they were in the original 
+   document.
+
+Dynamically Serialize Properties
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use a method to determine whether or not to serialize a property. For
+the driver to automatically use the method when serializing, the method must be
+named ``ShouldSerialize<property name>``. When the driver sees a method with
+this naming convention, it will use the method to determine whether or not to
+serialize properties of the provided property name.
+
+The following example creates a method that only serializes the ``Age`` property
+if its value is greater than ``0``. The driver ignores any properties whose
```

Comment:
"ignores any Age property values"

This only applies to serialization, not deserialization.

Also, normally this would be used to omit the default value, not a whole range of values (like negative ages). So I would expect to see `Age != 0` rather than `Age > 0`.

Of course if all ages are positive it amounts to the same thing. But I'd still rather see `!=` rather than `>`.

Reply:
Changed to `Age != 0` and specified that the driver only ignores during serialization